src/components/Form.js

```diff
@@ -25,13 +25,14 @@ export default class Form extends Component {
       status: "initial",
       formData,
       edit,
-      errors: edit ? this.validate(formData) : []
+      errors: edit ? this.validate(formData,props) : []
     };
   }
 
-  validate(formData) {
+  validate(formData,newProps) {
     const validator = new Validator();
-    return validator.validate(formData, this.props.schema).errors;
+    var props=!newProps?this.props:Object.assign({},this.props,newProps);
```

Comment:
I believe it would be easier to read if you do the conditionel on `newProps` directly, rather than negating it.